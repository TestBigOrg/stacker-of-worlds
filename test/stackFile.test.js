'use strict';

const fs = require('fs');
const path = require('path');
const stackFile = require('../lib/stackFile');

const getFixturePath = (name) => path.join(__dirname, 'fixtures', `${name}.geojson`);

const loadResult = (filepath) => {
  return new Promise((resolve, reject) => {
    fs.readFile(filepath, 'utf8', (err, data) => {
      if (err) return reject(err);
      resolve(JSON.parse(data));
    });
  });
};

describe('stackFile', () => {
  it('featurecollection-multiworld', () => {
    return stackFile(getFixturePath('featurecollection-multiworld')).then((filepath) => {
      expect(typeof filepath).toBe('string');
      return loadResult(filepath);
    }).then((actual) => {
      expect(actual).toMatchSnapshot();
    });
  });
});
