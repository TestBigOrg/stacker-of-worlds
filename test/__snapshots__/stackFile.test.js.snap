exports[`stackFile featurecollection-multiworld 1`] = `
Object {
  "features": Array [
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              Array [
                180,
                67.18620619898901,
              ],
              Array [
                179.8297119140625,
                67.28053044353817,
              ],
              Array [
                180,
                67.33770521826514,
              ],
              Array [
                180,
                67.18620619898901,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                180,
                67.9871533943935,
              ],
              Array [
                179.439697265625,
                67.96227048518728,
              ],
              Array [
                179.4287109375,
                68.17972379832914,
              ],
              Array [
                180,
                68.05806179295108,
              ],
              Array [
                180,
                67.9871533943935,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                180,
                67.72896346207641,
              ],
              Array [
                179.6484375,
                67.69277095059344,
              ],
              Array [
                179.58251953125,
                67.87554134672943,
              ],
              Array [
                180,
                67.8237630400101,
              ],
              Array [
                180,
                67.72896346207641,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                -180,
                67.33770521826514,
              ],
              Array [
                -179.8736572265625,
                67.38003710805398,
              ],
              Array [
                -179.89013671875,
                67.12515524355918,
              ],
              Array [
                -180,
                67.18620619898901,
              ],
              Array [
                -180,
                67.33770521826514,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                -180,
                68.05806179295108,
              ],
              Array [
                -179.72534179687503,
                67.99934114512352,
              ],
              Array [
                -180,
                67.9871533943935,
              ],
              Array [
                -180,
                68.05806179295108,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                -180,
                67.8237630400101,
              ],
              Array [
                -179.58251953124997,
                67.77186968146705,
              ],
              Array [
                -180,
                67.72896346207641,
              ],
              Array [
                -180,
                67.8237630400101,
              ],
            ],
          ],
        ],
        "type": "MultiPolygon",
      },
      "properties": Object {},
      "type": "Feature",
    },
    Object {
      "geometry": Object {
        "coordinates": Array [
          Array [
            Array [
              -69.60937499999999,
              -53.330872983017066,
            ],
            Array [
              150.46875,
              -27.68352808378776,
            ],
            Array [
              180,
              -21.991951190715152,
            ],
          ],
          Array [
            Array [
              -180,
              -21.991951190715152,
            ],
            Array [
              43.59375,
              24.206889622398023,
            ],
            Array [
              180,
              40.75212192228057,
            ],
          ],
          Array [
            Array [
              -180,
              40.75212192228057,
            ],
            Array [
              -71.015625,
              51.6180165487737,
            ],
            Array [
              180,
              64.3880331727298,
            ],
          ],
          Array [
            Array [
              -180,
              64.3880331727298,
            ],
            Array [
              -149.0625,
              65.65827451982659,
            ],
          ],
        ],
        "type": "MultiLineString",
      },
      "properties": Object {
        "id": "aaa",
      },
      "type": "Feature",
    },
  ],
  "type": "FeatureCollection",
}
`;
